path.go

```diff
@@ -197,7 +197,12 @@ func zoneFromPath(r *http.Request, rec record) (string, int, []string, error) {
 		path = path[:strings.Index(path, "/info/refs")]
 	}
 
-	path = fmt.Sprintf("%s?%s", path, r.URL.RawQuery)
+	// Only add request query to path if the custom regex needs it. Unless it
+	// might cause problems on custom regexes that don't need query to generate
+	// the zone
+	if strings.Contains(rec.Re, "query") {
```

Comment:
Couldn't we add it to the context instead? That way we have it accessible, but not pollute path not even in the case when it's needed by the regex.

Reply:
Not sure how adding it to the context can be helpful because either way, we have to merge path and query before passing it to the custom regex.

Reply:
Sidenote: I think it would be good to work on a refactor for the `zoneFromPath` function in one of the next milestones because that part of the code is really hard to maintain right now.
We can move some parts of it to separate methods and add more unit tests for them to make sure everything is stable and works as expected and make the futures changes easier to implement.